packages/neuron-wallet/src/startup/sync-block-task/init-database.ts

```diff
@@ -1,14 +1,24 @@
 import initConnection from '../../database/chain/ormconfig'
 import Utils from '../../services/sync/utils'
 import genesisBlockHash from './genesis'
+import { updateMetaInfo, getMetaInfo } from '../../database/chain/meta-info'
+import LockUtils from '../../utils/lock-utils'
 
 export const initDatabase = async () => {
   try {
     const hash = await genesisBlockHash()
     await initConnection(hash)
+    const systemScriptInfo = await LockUtils.systemScript()
+    updateMetaInfo({ genesisBlockHash: hash, systemScriptInfo })
   } catch (err) {
-    Utils.sleep(1000)
-    await initDatabase()
+    try {
+      const metaInfo = getMetaInfo()
+      await initConnection(metaInfo.genesisBlockHash)
+      LockUtils.setSystemScript(metaInfo.systemScriptInfo)
```

Comment:
Since `LockUtils` is not a pure utils class any more (it stores its own state), please move it to models or other appropriate module.

Reply:
Move to `models` and still named `LockUtils`